### DISCUSSION TITLE
How to disable the entire widget

### DISCUSSION THREAD

**Discussion by asiby** [sa-si-dev/virtual-select#81] [Q&A] [answered]
The Virtual Server widget has an option for specifying the disabled list options/items. However, in our use case, we need to disable the whole widget if some conditions are not met.

I am thinking that we could have a top level property called `disabled` not unlike the attribute of the same name that is available for any HTML form elements.

If this is already possible, then how is it done? If not, then the next question will be: is it an acceptable feature request? I will be willing to contribute some code to make it happen.

Thanks

**Comment by sa-si-dev** [comment]
@asiby, you are welcome to contribute to the plugin. Please implement it and submit a PR.

**Comment by sa-si-dev** [comment, marked as answer]
@asiby this feature has been implemented in the [release v1.0.21](https://github.com/sa-si-dev/virtual-select/releases/tag/v1.0.21)

```js
/** to disable all dropdown options */
document.querySelector('#sample-select').setDisabledOptions(true);

/** to disable dropdown */
document.querySelector('#sample-select').disable();
```
